core/src/main/java/org/elasticsearch/common/geo/GeoDistance.java

```diff
@@ -40,7 +41,26 @@
 
     /** Creates a GeoDistance instance from an input stream */
     public static GeoDistance readFromStream(StreamInput in) throws IOException {
+        Version clientVersion = in.getVersion();
         int ord = in.readVInt();
+        // bwc client deprecation for FACTOR and SLOPPY_ARC
+        if (clientVersion.before(Version.V_5_3_3)) {
+            switch (ord) {
+                case 0: return PLANE;
+                case 1:
+                    // bwc client deprecation for FACTOR
+                    DEPRECATION_LOGGER.deprecated("[factor] is deprecated. Using [plane] instead.");
+                    return PLANE;
+                case 2: return ARC;
+                case 3:
+                    // bwc client deprecation for SLOPPY_ARC
+                    DEPRECATION_LOGGER.deprecated("[sloppy_arc] is deprecated. Using [arc] instead.");
+                    return ARC;
+                default:
+                    throw new IOException("Unknown GeoDistance ordinal [" + ord + "]");
+            }
+        }
+
```

Comment:
Can you also fix the write side so that if the client is on 5.4 while the cluster is on 5.3 then things would still work?